core/backend/src/internal/IModelDbFontsImpl.ts

```diff
@@ -0,0 +1,220 @@
+import { IModelDb } from "../IModelDb";
+import { EmbedFontFileArgs, IModelDbFonts, QueryMappedFamiliesArgs } from "../IModelDbFonts";
+import { EmbeddedFontFile } from "./FontFileImpl";
+import { assert, Logger } from "@itwin/core-bentley";
+import type { IModelJsNative } from "@bentley/imodeljs-native";
+import { FontFile } from "../FontFile";
+import { BackendLoggerCategory } from "../BackendLoggerCategory";
+
+class IModelDbFontsImpl implements IModelDbFonts {
+  public readonly [_implementationProhibited] = undefined;
+  readonly #db: IModelDb;
+
+  public constructor(iModel: IModelDb) {
+    this.#db = iModel;
+  }
+
+  public queryMappedFamilies(args?: QueryMappedFamiliesArgs): Iterable<FontProps> {
+    const fontProps = this.#queryFontTable();
+    if (args?.includeNonEmbedded) {
+      return fontProps;
+    }
+
+    const fontNames = this.#getEmbeddedFontNames();
+    return fontProps.filter((x) => fontNames.includes(x.name));
+  }
+
+  public queryEmbeddedFontFiles(): Iterable<FontFile> {
+    const files: FontFile[] = [];
+    this.#db.withSqliteStatement(`SELECT Id,StrData FROM be_Prop WHERE Namespace="dgn_Font" AND Name="EmbeddedFaceData"`, (stmt) => {
+      while (DbResult.BE_SQLITE_ROW === stmt.step()) {
+        let faces;
+        try {
+          faces = JSON.parse(stmt.getValueString(1)) as IModelJsNative.FontFaceProps[];
+        } catch (e) {
+          Logger.logException(BackendLoggerCategory.IModelDb, e);
+        }
+
+        if (!Array.isArray(faces) || faces.length === 0) {
+          continue;
+        }
+
+        let type = faces[0].type;
+        if (type !== FontType.Rsc && type !== FontType.Shx) {
+          type = FontType.TrueType;
+        }
+
+        files.push(new EmbeddedFontFile(this.#db, stmt.getValueInteger(0), type, faces));
+      }
+    });
+
+    return files;
+  }
+
+  public async embedFontFile(args: EmbedFontFileArgs): Promise<void> {
+    this.#requireWritable();
+
+    if (!args.file.isEmbeddable) {
+      throw new Error("Font does not permit embedding");
+    }
+
+    const file = args.file;
+    for (const existing of this.queryEmbeddedFontFiles()) {
+      if (existing[_key] === file[_key]) {
+        // Already embedded - it's a no-op.
+        return;
+      }
+    }
+
+    let id = 0;
+    const codes = this.#db.codeService?.internalCodes;
+    if (codes) {
+      id = await codes.writeLocker.reserveEmbeddedFaceDataId(args.file[_key]);
+    } else {
+      // CodeService not configured - schema lock required to prevent conflicting Ids in be_Prop table.
+      await this.#db.acquireSchemaLock();
+      const sql = `SELECT MAX(Id) FROM be_Prop WHERE Namespace="dgn_Font" AND Name="EmbeddedFaceData"`;
+      id = this.#db.withSqliteStatement(sql, (stmt) => stmt.nextRow() ? stmt.getValueInteger(0) + 1 : 1);
+    }
+    
+    assert(id > 0);
+    const data = file[_getData]();
+    this.#db[_nativeDb].embedFontFile(id, file[_faceProps], data, true);
+
+    if (!args.dontAllocateFontIds) {
```

Comment:
Understanding you wanted to preserve allocation as default, did you consider a non-negative name for `dontAllocateFontIds`?

Reply:
Can you suggest one that you like better? `suppressFontIdAllocation`? (meh).

Reply:
Yeah it's tough :) `skipFontIdAllocation`?

Reply:
Do you love that enough to make me change it?

Reply:
I don't love it. I think it's better. I won't be heartbroken if you decide to leave as-is.

Reply:
I would have preferred `dontNotSkipFontIdAllocation` but ok. Pushed.